src/ControlSystem/Tags.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/OptionTags.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace control_system::Tags {
+/// The measurement timescales associated with
+/// domain::Tags::FunctionsOfTime.
+struct MeasurementTimescales : db::SimpleTag {
+  using type = std::unordered_map<
+      std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>;
+
+  static constexpr bool pass_metavariables = true;
+
+  template <typename Metavariables>
+  using option_tags =
+      tmpl::list<domain::OptionTags::DomainCreator<Metavariables::volume_dim>>;
+
+  template <typename Metavariables>
+  static type create_from_options(
+      const std::unique_ptr<::DomainCreator<Metavariables::volume_dim>>&
+          domain_creator) noexcept {
+    std::unordered_map<std::string,
+                       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+        timescales;
+    for (const auto& function_of_time : domain_creator->functions_of_time()) {
+      const double function_initial_time =
+          function_of_time.second->time_bounds()[0];
+      auto initial_time_scale = make_with_value<DataVector>(
+          function_of_time.second->func(function_initial_time)[0], 0.1);
```

Comment:
Why `0.1` here?

Also, I suggest you write `const DataVector used_for_size = function_of_time->func(function_initial_time)[0];` and pass it to `make_with_value`, just to make clear the values don't matter.

Reply:
0.1 was arbitrary since the actual control systems don't exist yet to set it.  Actually, though, setting it to the initial step size is less arbitrary and might even make sense as a real initial value.